### PR TITLE
BUG: Raise exception for inconsistent WAV header

### LINKED ISSUE #13988: An error occurred when using scipy.io.wavfile of scipy 1.6 version or higher to read the wav file
When I use scipy version 1.6 and above to read a single-channel wav file, the length of the signal is half of the original signal length. The reading of this signal is completely wrong. In order to verify whether it is a problem with this version of the library, I read the same wav file with a library below version 1.6 and matlab, and finally found that there is really some problem with this version of the code.

#### Reproducing code example:

```
import scipy.io.wavfile as wav
filePath = '1.wav'
fs, signalWav = wav.read(filePath)
print(len(signalWav))
```

#### Error message:

```
47488
(Actually the output should be 94976)
```

### PR DESCRIPTION
**Pull request by endolith**
#### Reference issue
Closes https://github.com/scipy/scipy/issues/13988

#### What does this implement/fix?
Files with a contradictory nBlockAlign field were still being opened,
but with the wrong interpretation of the data.

Example:

    nSamplesPerSec = 0x0000AC44 = 44100 Hz
    nAvgBytesPerSec = 0x00015888 = 88200 bytes per second
    nBlockAlign = 0x0004 = 4 bytes per sample

But nAvgBytesPerSec should equal nSamplesPerSec * nBlockAlign

### CODE REVIEW

File: scipy/io/tests/test_wavfile.py [hunk 1]
```diff
@@ -361,6 +361,15 @@ def test_read_incomplete_chunk():
                 wavfile.read(fp, mmap=mmap)
 
 
+def test_read_inconsistent_header():
+    # File header's size fields contradict each other
+    for mmap in [False, True]:
+        filename = 'test-8000Hz-le-3ch-5S-24bit-inconsistent.wav'
```
Comment (tupui) [inline scipy/io/tests/test_wavfile.py:367]: Couldn't we have a binary string instead of having the need to add a file?

File: scipy/io/wavfile.py [hunk 2]
```diff
@@ -384,6 +384,14 @@ def _read_fmt_chunk(fid, is_big_endian):
     # fmt should always be 16, 18 or 40, but handle it just in case
     _handle_pad_byte(fid, size)
 
+    if format_tag == WAVE_FORMAT.PCM:
+        if bytes_per_second != fs * block_align:
+            raise ValueError("WAV header is invalid: nAvgBytesPerSec must "
```
Comment (tupui) [inline scipy/io/wavfile.py:389]: Apparently now we ask the last space to be at the start of the line instead of being the last character.

**Review by tupui — APPROVED** [review]
Thanks, LGTM. I just have a question and a non blocking suggestion.

File: scipy/io/tests/test_wavfile.py (hunk 1 above)
Comment (endolith) [reply scipy/io/tests/test_wavfile.py:367]: I'm just following the example of all the other test files.  Probably good to test it the way users are going to, though?

File: scipy/io/wavfile.py (hunk 2 above)
Comment (tupui) [reply scipy/io/wavfile.py]: This is still in discussion there and would be a nice to have apparently #13955

File: scipy/io/tests/test_wavfile.py (hunk 1 above)
Comment (tupui) [reply scipy/io/tests/test_wavfile.py:367]: You're the expert here if it makes sense sure. I will merge once you confirmed.

Comment (endolith) [reply scipy/io/tests/test_wavfile.py:367]: Well it's the same as all the other examples, so if we're going to change it to strings, that should be done in another PR.

Comment (tupui) [reply scipy/io/tests/test_wavfile.py:367]: Ok, fine with me. I am merging then :) Thanks.

**Pull request merged**